btk/draw_blends.py

```diff
@@ -415,8 +415,10 @@ def render_single(self, entry, filt, psf, survey):
         Returns:
             gal (galsim.InterpolatedImage): The galsim profile of a single galaxy
         """
-        k = int(np.random.rand(1) * len(self.catalog))  # catalog_line["btk_index"][0]
-        gal = self.catalog.makeGalaxy(k, gal_type="real", noise_pad_size=0).withFlux(1)
+
+        galsim_catalog = self.catalog.get_galsim_catalog()
+        gal_flux = get_flux(entry['ref_mag'], filt, survey)
```

Comment:
This looks right to me, although I know there are several definitions of magnitudes floating around. Is this a AB magnitude? (I assume this is the standard convention for most things ? )

Reply:
I think this should be managed in the _prepare_table function in all cases, ref_mag is one of the BTK specific arguments. This should be the AB magnitude for the function to give the correct result though.

Reply:
do you know if it's an AB magnitude? @thuiop or @mpaillassa